backend/coins/eth/account.go

```diff
@@ -246,11 +246,24 @@ func (account *Account) updateOutgoingTransactions(tipHeight uint64) {
 	// Update the stored txs' metadata if up to 12 confirmations.
 	for _, tx := range outgoingTransactions {
 		remoteTx, err := account.coin.client.TransactionReceiptWithBlockNumber(context.TODO(), tx.Transaction.Hash())
-		if err != nil {
-			account.log.WithError(err).Error("could not fetch transaction")
-			continue
-		}
-		if remoteTx == nil {
+		if remoteTx == nil || err != nil {
+			// Transaction not found. This usually happens for pending transactions.
+			// In this case, check if the node actually knows about the transaction, and if not, re-broadcast.
+			// We do this because it seems that sometimes, a transaction that was broadcast without error still ends up lost.
+			_, _, err := account.coin.client.TransactionByHash(context.TODO(), tx.Transaction.Hash())
+			if err != nil {
+				tx.BroadcastAttempts++
+				account.log.WithError(err).Errorf("could not fetch transaction - rebroadcasting, attempt %d", tx.BroadcastAttempts)
+				if err := account.coin.client.SendTransaction(context.TODO(), tx.Transaction); err != nil {
```

Comment:
yes

Reply:
so, if someone accumulates say 10 transactions like this, they'll be potentially hitting etherscan at 5 QPS every 30 sec, assuming each request takes 200ms on average. it may get pretty bad at some point...